en/material-system/effect-syntax.md

```diff
@@ -0,0 +1,259 @@
+
+## Shader Chunks
+Syntactically shader chunks are a superset of GLSL, all the extended features will be processed immediately at resource compile-time.<br>
+It can either be written inside the `CCProgram` block in `.effect` file, or directly in a seperate `.chunk` file.
+We recommend limiting the syntax within GLSL version 300 ES range to maintain best portability,<br>
+although you can always write your own porting using GLSL built-in macro `__VERSION__`.<br>
+
+This section will introduce the 'domain-specific' extended features,<br>
+all of which will be processed immediately at resource compile-time.<br>
+Also feel free to look around the built-in effects, which are always excellent concrete examples.
+
+On top of GLSL, several c-style extensions are naturally introduced：
+
+### Chunk Include
+Just like the `include` directive in C/C++, you can include other shader chunks：
+```c
+#include <cc-global>
+#include "../headers/my-shading-algorithm.chunk"
+```
+Relevant details:
+* The `.chunk` extension can be omitted; quotation marks and angle brackets means the same;
+* Every included header is guaranteed to be expanded only once; so every module could (and in fact should) include all its dependencies, even if there are overlaps;
+* Dead code elimination is done at compile-time too, so including lots of unused utility functions shouldn't be of concern;
+* There are two ways to reference a external chunk file: using the relative path to current file ('relative path'), or the relative path to `assets/chunks` folder of current project ('project absolute path'). If the specified file interpreted under both rules are present, the latter is preferred;
+* When including files from other databases (like the internal database), only project absolute path is supported; when there are multiple databases have the same specified file, the priority is: User Project DB > Plugin DB > Internal DB;
+* The built-in chunks are located directly inside `assets/chunks` in internal database, so you can include these without path prefix;
+* All `CCProgram` blocks in the same effect file can include each other;
+
+### Pre-processing Macros
+Currently the effect system tends to take advantage of the language built-in pre-processing macros to create shader variants.<br>
+The effect compiler will collect the macros appeared in shaders, and declarations will be inserted accordingly at runtime.<br>
+So for the most part you can use them without thinking about the effect compiler,<br>
+while material inspector will automatically integrate both macros and shader properties into a natual editting interface.<br>
+Relevant details:
+* To type check as many branches as possible at effect compile-time, the strategy currently taken is to set all macros to `1` (or its given default value) first, then send it down the type check backend; so make sure this combination works (or if not, maybe you need numerical macros, specified in the next section);
```

Comment:
send it down the type check backend;
-->
This doesn't make sense to me what is meant here.